Kudu.Agent/Controllers/KuduProcessController.cs

```diff
@@ -0,0 +1,324 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Kudu.Agent.Contracts;
+using Kudu.Agent.Util;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Authentication;
+using System.Threading;
+
+namespace Kudu.Agent.Controllers
+{
+    [ApiController]
+    [Route("/processes")]
+    public class KuduProcessController : ControllerBase
+    {
+        [HttpGet]
+        public IActionResult GetAllProcesses()
+        {
+            IEnumerable<ProcessInfo> processes = Process.GetProcesses()
+                .Where(p => !AppServicePlatformProcess(p))
+                .Select(p => GetProcessInfo(p, details: false));
+
+            return Ok(processes);
+        }
+
+        [HttpGet("{id}")]
+        public IActionResult GetProcess(int id)
+        {
+            Process p = GetProcessById(id);
+            if (p == null)
+            {
+                return NotFound();
+            }
+
+            ProcessInfo info = GetProcessInfo(p, details: true);
+
+            return Ok(info);
+        }
+
+        [HttpDelete, HttpPost]
+        public IActionResult KillProcess(int id)
+        {
+            Process p = GetProcessById(id);
+            p.Kill(entireProcessTree: true);
+
+            return Ok();
+        }
+
+        [HttpGet("{id}/threads")]
+        public IActionResult GetAllThreads(int id)
+        {
+            Process process = GetProcessById(id);
+            IEnumerable<ProcessThreadInfo> threadsInfo = GetThreads(process);
+
+            return Ok(threadsInfo);
+        }
+
+        [HttpGet("{processId}/threads/{threadId}")]
+        public IActionResult GetThread(int processId, int threadId)
+        {
+            Process p = GetProcessById(processId);
+            IEnumerable<ProcessThread> threads = p.Threads.Cast<ProcessThread>();
+            ProcessThread thread = threads.FirstOrDefault(t => t.Id == threadId);
+            if (thread == null)
+            {
+                return NotFound();
+            }
+
+            ProcessThreadInfo threadInfo = GetProcessThreadInfo(thread, details: true);
+
+            return Ok(threadInfo);
+        }
+
+        [HttpGet("{id}/modules")]
+        public IActionResult GetAllModules(int id)
+        {
+            Process process = GetProcessById(id);
+            IEnumerable<ProcessModuleInfo> modulesInfo = GetModules(process);
+
+            return Ok(modulesInfo);
+        }
+
+        [HttpGet("{id}/modules/{baseAddress}")]
+        public IActionResult GetModule(int id, string baseAddress)
+        {
+            Process p = GetProcessById(id);
+            IEnumerable<ProcessModule> modules = p.Modules.Cast<ProcessModule>();
```

Comment:
When GetProcessById returns null please handle this null condition in the callers of this condition otherwise it would cause null pointer exception.